src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -190,7 +219,11 @@ public Revision sortBranchesForRevision(Revision revision, List<BranchSpec> bran
                         }
 
                     } finally {
-                        walk.release();
+                        try {
+                            walk.release();
```

Comment:
I think you're saying that there are cases where the call to walk.release() may not even be reached because there was a LinkageError while processing the byte code.  Do I need to switch to use reflection to find the walk.release() method in addition to the current use of reflection to find the walk.close() method?

I've not seen a LinkageError in my testing, but my testing is clearly not exhaustive, since I didn't see this problem weeks or a month after the git-client-plugin change to JGit 4 was made on the git client plugin 2.0.0-beta branch.